dvc/command/plots.py

```diff
@@ -34,47 +37,78 @@ def run(self):
                 return 1
 
         try:
-            plots = self._func(targets=self.args.targets, props=self._props())
+            onerror = Onerror()
+            plots = self._func(
+                targets=self.args.targets, props=self._props(), onerror=onerror
+            )
 
             if self.args.show_vega:
                 target = self.args.targets[0]
                 logger.info(plots[target])
                 return 0
 
-            rel: str = self.args.out or "plots.html"
-            path = (Path.cwd() / rel).resolve()
-            self.repo.plots.write_html(
-                path, plots=plots, html_template_path=self.args.html_template
-            )
-
         except DvcException:
             logger.exception("")
             return 1
 
-        assert path.is_absolute()  # as_uri throws ValueError if not absolute
-        url = path.as_uri()
-        logger.info(url)
-        if self.args.open:
-            import webbrowser
+        return_value = 0
+        if onerror.any_failed():
+            self._log_errors(onerror)
+            return_value = 1
 
-            opened = webbrowser.open(rel)
-            if not opened:
-                logger.error("Failed to open. Please try opening it manually.")
-                return 1
+        if plots:
+            rel: str = self.args.out or "plots.html"
+            path = (Path.cwd() / rel).resolve()
+            self.repo.plots.write_html(
+                path, plots=plots, html_template_path=self.args.html_template
+            )
 
-        return 0
+            assert (
+                path.is_absolute()
+            )  # as_uri throws ValueError if not absolute
+            url = path.as_uri()
+            logger.info(url)
+            if self.args.open:
+                import webbrowser
+
+                opened = webbrowser.open(rel)
+                if not opened:
+                    logger.error(
+                        "Failed to open. Please try opening it manually."
+                    )
+                    return_value = 1
+        else:
+            from dvc.ui import ui
+
+            ui.warn(
```

Comment:
We used to create file with no actual plots. After this change we will not create it if there are no plots.

Reply:
@pared, we haven't yet decided on colors yet, so we are using `ui.error_write` in other places. It does look good, so maybe that's okay, but just wanted to let you know.